http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/BaseCloudServiceRuntimeMonitor.scala

```diff
@@ -320,20 +320,36 @@ abstract class BaseCloudServiceRuntimeMonitor[F[_]] {
           )
       }
       // wait for 10 minutes for tools to start up before time out.
-      availableTools <- streamFUntilDone(checkTools, 120, 5 seconds).compile.lastOrError
+      availableTools <- streamFUntilDone(checkTools, monitorConfig.checkToolsMaxAttempts, monitorConfig.checkToolsDelay).compile.lastOrError
       r <- availableTools match {
         case a if a.forall(_._2) =>
           readyRuntime(runtimeAndRuntimeConfig, ip, monitorContext, dataprocInstances)
         case a =>
-          val toolsStillNotAvailable = a.collect { case x if x._2 == false => x._1 }
-          failedRuntime(
-            monitorContext,
-            runtimeAndRuntimeConfig,
-            Some(
-              RuntimeErrorDetails(s"${toolsStillNotAvailable} didn't start up properly", None, Some("tool_start_up"))
-            ),
-            dataprocInstances
-          )
+          // If the tools didn't start up, resolve the DB status one more time.
+          // It's possible that the user explicitly deleted the runtime in which case
+          // we can ignore the error. Otherwise, move the runtime to Error status.
+          for {
+            curStatusOpt <- clusterQuery.getClusterStatus(runtimeAndRuntimeConfig.runtime.id).transaction
+            curStatus <- F.fromOption(
+              curStatusOpt,
+              new Exception(s"Cluster with id ${runtimeAndRuntimeConfig.runtime.id} not found in the database")
+            )
+            res <- curStatus match {
+              case RuntimeStatus.Deleted => F.pure[CheckResult](((), None))
```

Comment:
shall we add the specific case in `failedRuntime` function instead of here? That way, no matter where we end up with `Deleted` status, `failedRuntime` won't try to put the cluster in `Error` status

Reply:
Hmm.. yeah that probably makes sense

Reply:
I also notice that `GceRuntimeMonitor.failedRuntime` and `DataprocRuntimeMonitor.failedRuntime` are very similar.. the only difference is how errors are persisted to the DB. Considering moving some of this logic to `BaseCloudServiceRuntimeMonitor` to avoid duplication.